src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Legacy/LegacySyntaxNodeExtensions.cs

```diff
@@ -341,6 +341,7 @@ public static IEnumerable<SyntaxNode> FlattenSpans(this SyntaxNode node)
         }
     }
 
+    [Obsolete("This API does not have good performance, and should be avoided")]
```

Comment:
Feels like we should fix the perf of this method, rather than introduce a new workaround?

Reply:
The only remaining usage of it is in the legacy editor. I can file a follow up if you like (it would be in `area-razor.compiler` 😛) but I don't think its worth blocking this PR on.

Reply:
Well, my point is that it feels like the wrong solution to have the new editor move off this API onto a workaround outside the compiler layer.

Reply:
IMO the only reason `TryGetPreviousSibling` isn't in the compiler layer is because it was written before the repo merge. I can certainly move it if you agree? I don't consider it to be a workaround for a perf problem, just a better API for what the code is trying to do. I hope the "WorkaroundQuirks" method that calls it can be removed with your new find owner method.

Reply:
I tend to agree with @333fred. Can we just make `PreviousSpan()` internally call `GetOutermostNode()` and `TryGetPreviousSibling()`?

Reply:
I did try updating the remaining call site and the tests failed... but actually, I might have forgotten to call GetOutermostNode, so let me double check.

Reply:
Oh no, I didn't update this, because `TryGetPreviousSibling` is not in a layer that makes it usable. Okay, will try again properly this time, and move it :)

Reply:
Nope, this won't work. The brace indenter really does want the inner most previos span, which is exactly what the `PreviousSpan` method is doing. The perf fix is just to not use it in the tooling layer, where we don't actually want that behaviour.

eg, in the test it is trying to find the previous span of the highlighted `CSharpStatementLiteral`, which is not any kind of sibling.
<img width="639" alt="image" src="https://github.com/dotnet/razor/assets/754264/17fa6433-7fd9-4d44-a8b8-9c8bb9b9d955">

Reply:
If that's the case, then perhaps this shouldn't be obsoleted?

Reply:
I feel like obsolete is the perfect scenario: It's still in use, so can't be removed, but only in use by an obsolete(ish) editor. We want to avoid any new code being written that calls it. Being so low down in the stack (from a tooling perspective at least), its a tempting site in a completion list.

Reply:
But it doesn't seem broken? The work it does is the precise work that needs to be done to get the previous span. If that's not what the IDE actually needs, then it seems fine not to call it. But that doesn't mean we should obsolete it with a different method that doesn't cover what this method does.

Reply:
Okay, will revert the obsoletion